docs/current/cookbook/snippets/cache-dependencies/main.py

```diff
@@ -0,0 +1,43 @@
+import sys
+
+import anyio
+
+import dagger
+
+
+async def main():
+    config = dagger.Config(log_output=sys.stdout)
+
+    async with dagger.Connection(config) as client:
+        # use a python:3.11 container
+        # mount the source code directory on the host
+        # at /src in the container
+        # mount the cache volumes to persist dependencies
+        source = (
+            client.container()
+            .from_("python:3.11")
+            .with_directory(
+                "/src",
+                client.host().directory("."),
+                exclude=[".venv/", ".cache/", "ci/"],
+            )
+            .with_workdir("/src")
+            .with_mounted_cache(
+                "/root/.cache/pip", client.cache_volume("pip-python-311-myapp-myenv")
```

Comment:
That's quite long for a key 🙂 This isn't like `node_modules`, that would be more `.venv`. This cache should be ok to be global (scoped only to a python version) as it refers to the download of python packages.

Reply:
Modified